test/test_spectral_ops.py

```diff
@@ -21,6 +21,64 @@
 fft_fn = torch.fft
 import torch.fft
 
+
+def _complex_stft(x, *args, **kwargs):
+    # Transform real and imaginary components separably
+    stft_real = torch.stft(x.real, *args, **kwargs, return_complex=True, onesided=False)
+    stft_imag = torch.stft(x.imag, *args, **kwargs, return_complex=True, onesided=False)
+    return stft_real + 1j * stft_imag
+
+
+def _hermitian_conj(x, dim):
+    out = torch.empty_like(x)
+    mid = (x.size(dim) - 1) // 2
+    idx = [slice(None)] * out.dim()
+    idx_center = list(idx)
+    idx_center[dim] = 0
+    out[idx] = x[idx]
+
+    idx_neg = list(idx)
+    idx_neg[dim] = slice(-mid, None)
+    idx_pos = idx
+    idx_pos[dim] = slice(1, mid + 1)
+
+    out[idx_pos] = x[idx_neg].flip(dim)
+    out[idx_neg] = x[idx_pos].flip(dim)
+    if (2 * mid + 1 < x.size(dim)):
+        idx[dim] = mid + 1
+        out[idx] = x[idx]
+    return out.conj()
+
+
+def _complex_istft(x, *args, **kwargs):
+    # Decompose into Hermitian (FFT of real) and anti-Hermitian (FFT of imaginary)
+    n_fft = x.size(-2)
+    slc = (Ellipsis, slice(None, n_fft // 2 + 1), slice(None))
+
+    hconj = _hermitian_conj(x, dim=-2)
+    x_hermitian = (x + hconj) / 2
+    x_antihermitian = (x - hconj) / 2
+    istft_real = torch.istft(x_hermitian[slc], *args, **kwargs, onesided=True)
+    istft_imag = torch.istft(-1j * x_antihermitian[slc], *args, **kwargs, onesided=True)
+    return istft_real + 1j * istft_imag
+
+
+def stft_definition(x, hop_length, window):
```

Comment:
Comment for this function. Is there a better name for this function? Also, should it start with an underscore?

Reply:
"_stft_reference"?